lua/ulx/modules/sh/uptime.lua

```diff
@@ -0,0 +1,36 @@
+local function plural(value, noun)
+	return value ~= 1 and noun .. "s" or noun
+end
+
+local startTime = SysTime()
+local uptimeCommand = ulx.command("Chat", "ulx uptime", function()
+	local time_data = string.FormattedTime(SysTime() - startTime)
+	local h, m, s = time_data.h, time_data.m, time_data.s
+
+	if h < 1 and m < 1 then
+		ulx.fancyLog(
+			"Server has been running for #i #s",
+			s,
+			plural(s, "second")
```

Comment:
Given that `plural` is just defined in this file and also has context of the value, would it be better to make it `formatDuration(value, unit)` (or equivalent) to cut down on a bit of duplication here?

In fact, to _really_ cut down on duplication we could instead do something like...
```lua
local function formatDurationUnit(value, unit)
  return string.format("%d %s", value, value == 1 and unit or unit .. "s")
end

local function formatDuration(hours, minutes, seconds)
  if hours < 1 and minutes < 1 then
    return formatDurationUnit(seconds, "second")
  end

  ...
end
```

....and then we only need one `ulx.fancyLog` with a single source of truth for how the message is formatted

What do you think? (not too keen on "`formatDurationUnit`" but naming things is hard)

Reply:
This will break colored integers (see the screenshot).
I will have to return 2 values from the function and store them in local values and pass them to the `fancyLog` anyway, because of Lua's inability to handle passing multiple varargs.

```lua
local function duration(v)
	return v, "unit"
end
print(duration(1), duration(2)) --> 1, 2, unit
```

U sure u want that?

Reply:
I can just display the time in `%H:%M:%S` format if that bothers you.

Reply:
> This will break colored integers

I'm personally not keen on colouring the units individually anyway as it separates them from each other, but it's only minor and just my opinion

> I can just display the time in `%H:%M:%S` format if that bothers you.

Is there anything particularly bad about this? It cuts down on a lot of complexity for what would otherwise be a simple command, and seems marginally more useful by always showing the full elapsed time

Reply:
Although I think GitHub is having a stroke right now, because the commit isn't coming through.